stream-chat-android-client/src/main/java/io/getstream/chat/android/client/utils/observable/Subscriptions.kt

```diff
@@ -32,7 +33,7 @@ internal open class SubscriptionImpl(
 
         if (filter(event)) {
             try {
-                listener!!.invoke(event)
+                listener!!.onEvent(event)
```

Comment:
Why don't we safe unwrap here?

Reply:
Why don't we use a non-null property and assign a "no-op function" when it is disposed?

Reply:
We know that `listener` is never `null` here, otherwise the instance would also be already marked as disposed (in which case, `onNext` should never be called on it in the first place).

Reply:
If we know it's not null, what's the `try` block for?

Reply:
In the case when delivering the event throws (something errors out during the execution of the listener), we still want to run the after-event callback. We use it for the single-event subscribe mechanism, mostly. This way if that single delivery fails, disposal of the subscription still happens, and we don't end up delivering the next event as well.